lib/tty/pager/system.rb

```diff
@@ -123,22 +124,57 @@ def initialize(**options)
       #
       # @api public
       def page(text, &_callback)
-        return text unless output.tty?
+        write(text)
+        wait
+      end
+
+      # Spawn the pager process
+      #
+      # @return [PagerIO]
+      #   A wrapper for the external pager
+      #
+      # @api private
+      def start
+        # In case there's a previous pager running:
+        wait
 
         command = pager_command
         out = self.class.run_command(command)
         # Issue running command, e.g. unsupported flag, fallback to just command
-        if !out.empty?
+        unless out.empty?
           command = pager_command.split.first
         end
 
-        pager_io = open("|#{command}", 'w')
-        pid      = pager_io.pid
+        PagerIO.new(command)
+      end
 
-        pager_io.write(text)
-        pager_io.close
+      # Send text to the pager process. Starts a new process if it hasn't been
+      # started yet. Returns false if the pager was closed.
+      #
+      # @return [Boolean]
+      #   the success status of writing to the pager process
+      #
+      # @api public
+      def write(text)
```

Comment:
Couple things to make the API more flexible. First, `write` could accept any number of strings? Second, we could add `#<<` alias to mimic IO writing?

Reply:
Hm, I didn't know the built-in `write` accepts multiple strings. It's a bit weird to me since it seems like you could always concatenate them, but I suppose it would be less efficient. I'll probably do that tomorrow.

I implemented `<<` as an alias, but I'm not sure it's a great idea. The problem is that `pager << "foo" << "bar"` could only work if that method returns `self`, but that method currently returns true or false to indicate if the pager was closed. Instead of a boolean result, it could raise an error, but it would make handling that case a bit more annoying.